Expand dns alias security considerations (#2530)
httpwg/http-extensions: draft-ietf-httpbis-alias-proxy-status.md

The `next-hop-aliases` parameter does not include any DNSSEC information or imply that DNSSEC was used.
The information included in the parameter can only be trusted to be valid insofar as the client
trusts the proxy to provide accurate information. This information is intended to be used as
a hint, and SHOULD NOT be used for making security decisions about the identity of a resource accessed
through the proxy.

Inspecting CNAME chains can be used to detect cloaking of trackers or malicious hosts. However, the

> CNAME records could be omitted by a recursive resolver that is trying to hide this form of cloaking.

Reviewer: ```suggestion
CNAME records could be omitted by a recursive or authoritative resolver that is trying to hide this form of cloaking.
```

> Such attacks to avoid cloaking detection apply equally to clients directly performing DNS name
> resolution and proxies performing DNS name resolution on behalf of client. A malicious proxy could

Reviewer: ```suggestion
In particular, recursive or authoritative resolvers can omit these records for both clients directly performing DNS name
resolution and proxies performing DNS name resolution on behalf of client. A malicious proxy could
```

> also choose to not report these CNAME chains in order to hide the cloaking. In general, client can

Reviewer: ```suggestion
also choose to not report these CNAME chains in order to hide the cloaking. In general, clients can
```
